src/interfaces/ffi/hybrid_cc_aes.rs

```diff
@@ -493,55 +481,50 @@ pub unsafe extern "C" fn h_aes_decrypt_header_using_cache(
         return 1;
     };
 
-    let header: ClearTextHeader<Aes256GcmCrypto> =
-        ffi_unwrap!(decrypt_hybrid_header::<Aes256GcmCrypto>(
-            &cache.user_decryption_key,
-            encrypted_header_bytes
-        ));
+    // Decrypt header
+    let encrypted_header = ffi_unwrap!(EncryptedHeader::try_from_bytes(encrypted_header_bytes));
+    let header = ffi_unwrap!(encrypted_header.decrypt(&CC::default(), &cache.usk,));
 
     // Symmetric Key
     let allocated = *symmetric_key_len;
     let symmetric_key_bytes = header.symmetric_key.to_bytes();
-    let len = symmetric_key_bytes.len();
-    *symmetric_key_len = len as c_int;
-    if (allocated as usize) < len {
+    *symmetric_key_len = symmetric_key_bytes.len() as c_int;
+    if allocated < *symmetric_key_len {
         ffi_bail!(
             "The pre-allocated symmetric key buffer is too small; need {} bytes",
-            len
+            *symmetric_key_len
         );
     }
-    std::slice::from_raw_parts_mut(symmetric_key_ptr as *mut u8, len)
+    std::slice::from_raw_parts_mut(symmetric_key_ptr as *mut u8, symmetric_key_bytes.len())
         .copy_from_slice(&symmetric_key_bytes);
 
     // UID - if expected
     if !uid_ptr.is_null() && *uid_len > 0 {
         let allocated = *uid_len;
-        let uid_bytes = &header.meta_data.uid;
-        let len = uid_bytes.len();
-        *uid_len = len as c_int;
-        if (allocated as usize) < len {
+        *uid_len = header.metadata.uid.len() as c_int;
+        if allocated < *uid_len {
             ffi_bail!(
                 "The pre-allocated uid buffer is too small; need {} bytes",
-                len
+                *uid_len
             );
         }
-        std::slice::from_raw_parts_mut(uid_ptr as *mut u8, len).copy_from_slice(uid_bytes);
+        std::slice::from_raw_parts_mut(uid_ptr as *mut u8, header.metadata.uid.len())
+            .copy_from_slice(&header.metadata.uid);
     }
 
     // additional data - if expected
     if !additional_data_ptr.is_null() && *additional_data_len > 0 {
         let allocated = *additional_data_len;
-        let additional_data_bytes = &header.meta_data.additional_data;
-        if let Some(ad) = additional_data_bytes {
-            let len = ad.len();
-            *additional_data_len = len as c_int;
-            if (allocated as usize) < len {
+        if let Some(ad) = header.metadata.additional_data {
```

Comment:
Not a big fan of the `ad` abbreviation, it's not standard and not really readable I think…